uPortal-groups/uPortal-groups-core/src/test/java/org/apereo/portal/groups/RDBMSEntityGroupStoreTest.java

```diff
@@ -21,181 +29,263 @@
-            when(rs.getInt(3)).thenReturn(1);
-            when(rs.getString(4)).thenReturn("ignore-me");
-            when(rs.getString(5)).thenReturn("ignore-me");
-
-            Collection collection = new ArrayList();
-            Element el = new Element(cacheKey, collection.iterator());
-            when(parentGroupEntityGroupCache.get(cacheKey)).thenReturn(null, el, el, el);
-
+        String cacheKey = entityLocalKey + ":0:" + serviceName;
+        try (MockStaticUtil util = new MockStaticUtil()) {
             // for the purposes of this test, we don't care what is actually returned
             // from the database; we're just confirming that the cache is being
             // accessed the correct number of times
-            RDBMEntityGroupStore store = new RDBMEntityGroupStore();
+            store = new RDBMEntityGroupStore();
             store.findParentGroups(entityGroup);
             store.findParentGroups(entityGroup);
             store.findParentGroups(entityGroup);
             store.findParentGroups(entityGroup);
             verify(parentGroupEntityGroupCache, times(4)).get(cacheKey);
-            verify(parentGroupEntityGroupCache, times(1)).put(el);
+            verify(parentGroupEntityGroupCache, times(1)).put(new Element(cacheKey, null));
+        }
+    }
+
+    private void verifyResults(Iterator<IEntityGroup> iterator, List<IEntityGroup> expectedGroups) {
+        assertNotNull(iterator);
+        for (IEntityGroup expectedGroup : expectedGroups) {
+            assertTrue(iterator.hasNext());
+            IEntityGroup group = iterator.next();
+            assertNotNull(group);
+            assertEquals(expectedGroup, group);
+        }
+        assertFalse(iterator.hasNext());
+    }
+
+    private void assertIteratorDoesNotSupportRemove(Iterator<IEntityGroup> iterator) {
+        try {
+            iterator.remove();
+            fail();
+        } catch (UnsupportedOperationException expected) {
+        }
+    }
+
+    private void setUpCaching() {
+        groupSearchCache = spy(createNewCache(SEARCH_CACHE_NAME));
+        parentGroupEntityCache = spy(createNewCache(PARENT_GROUP_BY_ENTITY_CACHE_NAME));
+        parentGroupEntityGroupCache = spy(createNewCache(PARENT_GROUP_BY_ENTITY_GROUP_CACHE_NAME));
+        cacheManager = new CacheManager();
+        cacheManager.addCache(groupSearchCache);
+        cacheManager.addCache(parentGroupEntityCache);
+        cacheManager.addCache(parentGroupEntityGroupCache);
+        when(context.getBean("cacheManager", CacheManager.class)).thenReturn(cacheManager);
+    }
+
+    private Cache createNewCache(String cacheName) {
+        return new Cache(cacheName, 100, false, false, 300, 300);
+    }
+
+    private void setUpRdbms() throws Exception {
+        when(conn.prepareStatement(anyString())).thenReturn(ps);
+        when(ps.executeQuery()).thenReturn(rs);
+        when(rs.next()).thenReturn(true, false);
+        // set to null in order to avoid more mocks in
```

Comment:
I don't understand this comment. What is being set to `null` ?

Reply:
Ah, I see ...original code was 
```
            // set to null in order to avoid more mocks in
            // RDBMEntityGroupStore.instanceFromResultSet since we don't care about the results here
            when(rs.getString(1)).thenReturn(null);
```

I refactored to use actual cache instead of mock, so I had to set an actual ID instead of null.